### PR TITLE
[MAINTENANCE] Add tests for Common package

### LINKED ISSUE #781: [FUND] Add tests and significantly improve test coverage
## Description

Currently Kitodo.Presentation barely has any (unit) tests, so implementing new features and making sure that nothing broke is very tedious and time consuming - not only for developers but for the release management as well. This proposal suggests to write a lot of tests and significantly improve test coverage for Kitodo.Presentation.

## Related Issues

None.

## Expected Benefits of this Development

Test-driven development ensures that developing new features, updating components or re-factoring code doesn't break anything (or at least makes obvious what broke). This is especially important for older projects with a large and complex code base, since it's impossible for a single developer to know every constraint and dependency.
Having tests would therefor improve code quality, make development easier and help testing and reviewing pull requests for the release management.

## Estimated Costs and Complexity

Writing tests is not a complex, but a rather time-consuming task, so I'd estimate this to be of **high** cost and **low** complexity.

### PR DESCRIPTION
**Pull request by oliver-stoehr**

### CODE REVIEW

File: Tests/Fixtures/Common/solrcores.xml
Comment (sebastian-meyer) [inline Tests/Fixtures/Common/solrcores.xml:1]: This file is superseded by `Tests/Fixtures/Common/solrcores.csv`, isn't it? Then it should be removed from the PR.

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]

File: Tests/Functional/Common/SolrSearchTest.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        $this->setUpData(self::$databaseFixtures);
+        $this->solr = $this->setUpSolr(4, 0, self::$solrFixtures);
+    }
+
+    /**
+     * @test
+     */
+    public function canPrepareAndSubmit()
+    {
+        $this->markTestSkipped('Does not work in combination with other tests.');
```
Comment (sebastian-meyer) [inline Tests/Functional/Common/SolrSearchTest.php]: Hm, a skipped test is quite useless... There should be a way to make this work even in combination with the other tests. Where exactly does the test fail?

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]

File: Tests/Fixtures/Common/solrcores.xml
Comment (oliver-stoehr) [reply Tests/Fixtures/Common/solrcores.xml:1]: Thank you for pointing out, I removed this file.

File: Tests/Functional/Common/SolrSearchTest.php (hunk 1 above)
Comment (oliver-stoehr) [reply Tests/Functional/Common/SolrSearchTest.php]: I reenabled the test. It fails on line 55 at the first `assert` using the `SolrSearch` object from line 53. I could not find any difference in any variables accesses in the tested `SolrSearch::prepare` method.
I also switched to a different Solr core, but that didn't change the result.
However I'm not sure about the `DocumentRepository` passed to the constructor of the `SolrSearch` – I'm don't fully understand how the `DocumentRepository` object is created (or maybe reused?).

**Comment by sebastian-meyer** [comment]
I fixed the PHPStan issue in `master`. It should disappear as soon as you update your branch.

File: Tests/Functional/Common/SolrSearchTest.php (hunk 1 above)
Comment (sebastian-meyer) [reply Tests/Functional/Common/SolrSearchTest.php]: I am sorry, but I can't help you with that. Maybe @beatrycze-volk can give more insight into using the `DocumentRepository` correctly?

Comment (beatrycze-volk) [reply Tests/Functional/Common/SolrSearchTest.php]: > I am sorry, but I can't help you with that. Maybe @beatrycze-volk can give more insight into using the `DocumentRepository` correctly?

Actually more suitable person to answer this question would be @chrizzor as he has implemented this class.

**Comment by oliver-stoehr** [comment]
I fixed the test. It looks like some database query were cached, that were necessary to determine the correct solr core. The solr cores are now determined more directly.

**Review by sebastian-meyer — APPROVED** [review]
Great job! Thank you very much!

**Comment by sebastian-meyer** [comment]
Now you only need to update your branch to master and then I can merge it!

**Pull request merged**